### PR TITLE
Rad 2751/specify ad units set targeting for ast

### PR DESCRIPTION
**Pull request by sumit116**
## Type of change

-  Bugfix

## Description of change
Added Unit tests for setTargetingForAst()

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

### CODE REVIEW

File: test/spec/unit/core/targeting_spec.js [hunk 1]
```diff
@@ -326,4 +326,46 @@ describe('targeting tests', function () {
...
+      expect(window.apntag.setKeywords.called).to.equal(true);
+    });
+
+    it('should set array of addUnit codes', function() {
+      let adUnitCodes = ['testdiv1-abc-ad-123456-0', 'testdiv2-abc-ad-123456-0']
+      mockGetAllTargeting(adUnitCodes);
+      targetingInstance.setTargetingForAst(adUnitCodes);
+      expect(targetingInstance.getAllTargeting.called).to.equal(true);
+      expect(targetingInstance.resetPresetTargetingAST.called).to.equal(true);
+      expect(window.apntag.setKeywords.called).to.equal(true);
```
Comment (jaiminpanchal27) [inline test/spec/unit/core/targeting_spec.js]: To differentiate between this test and the one with single adUnitCode, you should test here that `window.apntag.setKeywords` executed twice. Also another assertion can be called with args. Check this https://github.com/sinonjs/sinon/issues/953#issuecomment-265447178

File: test/spec/unit/core/targeting_spec.js [hunk 2]
```diff
@@ -326,4 +326,46 @@ describe('targeting tests', function () {
...
+      sandbox.stub(targetingInstance, 'resetPresetTargetingAST').callsFake(function() {});
+      sandbox.stub(window.apntag, 'setKeywords').callsFake(function() {});
+    });
+    afterEach(function () {
+      sandbox.restore();
+    });
+
+    function mockGetAllTargeting(adUnitCodes) {
+      adUnitCodes = Array.isArray(adUnitCodes) ? adUnitCodes : Array(adUnitCodes);
+      sandbox.stub(targetingInstance, 'getAllTargeting').callsFake(function() {
```
Comment (jaiminpanchal27) [inline test/spec/unit/core/targeting_spec.js]: I would suggest here to use `stub.returns`. Changing the implementation of getAllTargeting with this fake function is not a good idea. So you can do something like below in each test case
```
targetingStub.returns({
   'adUnitCode': {hb_bidder:'appnexus'}
});
```

File: test/spec/unit/core/targeting_spec.js (hunk 1 above)
Comment (sumit116) [reply test/spec/unit/core/targeting_spec.js]: updated the tests.

File: test/spec/unit/core/targeting_spec.js (hunk 2 above)
Comment (sumit116) [reply test/spec/unit/core/targeting_spec.js]: Done.

**Comment by sumit116** [comment]
Docs PR: https://github.com/prebid/prebid.github.io/pull/1357

**Pull request merged**